### PR TITLE
FUSETOOLS2-1168 - Avoid override on startup of the kamel binary when

### PR DESCRIPTION
**Pull request by apupier**
auto-upgrade is off and provided kamel version is not a community
released version

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

### CODE REVIEW

File: src/test/suite/versionUtils.test.ts [hunk 1]
```diff
@@ -18,69 +18,101 @@
...
+		async function invalidateVersion(tagName: string, platform: Platform): Promise<void> {
+			try {
+				await versionUtils.getDownloadURLForCamelKTag(tagName, platform);
+				should.fail(`Downloading invalid Camel-K version (${tagName}) did not fail!`);
+			} catch (error) {
+				should.exist(error);
+			}
 		}
-	}
+	});
```
Comment (apupier) [inline src/test/suite/versionUtils.test.ts:91]: all this part is just that it has been grouped inside a "sub" test suite

**Comment by apupier** [comment]
the problem with current state is the case: auto-upgrade off, no valid tools path provided : it doesn't trigger a download

File: src/versionUtils.ts [hunk 2]
```diff
@@ -47,39 +47,44 @@ export async function testVersionAvailable(versionToUse: string): Promise<boolea
...
-			return false;
-		}
-	} else if (!versionToUse) {
-		const latestversion = await getLatestCamelKVersion();
-		if (failed(latestversion)) {
-			console.error(latestversion.error);
-			return true;
+	const runtimeVersionSetting = config.getKamelRuntimeVersionConfig();
+	const autoUpgradeSetting: boolean | undefined = config.getKamelAutoupgradeConfig();
+	const kamelCliToolsPath: string | undefined = config.getActiveKamelconfig();
```
Comment (lhein) [inline src/versionUtils.ts:52]: Just a minor thing but in the 2 lines above you are using the term `Setting` in the variable name while in this line you don't. Why not using the same terminology all over?

File: src/versionUtils.ts [hunk 3]
```diff
@@ -47,39 +47,44 @@ export async function testVersionAvailable(versionToUse: string): Promise<boolea
...
+		if (versionToUse) {
+			const passedVersionAvailable = await testVersionAvailable(versionToUse);
+			if (!passedVersionAvailable) {
+				const msg: string = `Camel K CLI Version ${versionToUse} unavailable.`;
+				extension.shareMessageInMainOutputChannel(msg);
+				return false;
+			}
+		} else if (!versionToUse) {
+			const latestversion = await getLatestCamelKVersion();
+			if (failed(latestversion)) {
```
Comment (lhein) [inline src/versionUtils.ts:64]: I am not a friend of naming a function `failed`. It does say nothing about what the function does.

Comment (apupier) [reply src/versionUtils.ts:64]: This was [introduced a while ago](https://github.com/camel-tooling/vscode-camelk/pull/148) and not modified in this PR. i reported https://issues.redhat.com/browse/FUSETOOLS2-1175

File: src/test/suite/versionUtils.test.ts (hunk 1 above)
Comment (apupier) [reply src/test/suite/versionUtils.test.ts:90]: I recommend to use "hide whitespace change" to hide this kind of change when reviewing this PR

**Review by lhein — APPROVED** [review]

**Review by bfitzpat — APPROVED** [review]

**Pull request merged**